### ISSUE TITLE
How to cycle buffers

### ISSUE THREAD

**Issue by bremner** [parkouss/webmacs#50]
On conkeror I was very used to M-p, M-n to flip through buffers (even though these are not standard emacs bindings). Is there a good alternative already for webmacs? If not, consider this a feature request.

**Comment by parkouss** [comment]
Sounds good! There is nothing in webmacs so far, so yes it is a feature request.

Note the implementation of switch-buffer should be reused for that I think, https://github.com/parkouss/webmacs/blob/master/webmacs/commands/webbuffer.py#L119.

**Comment by parkouss** [comment]
So currently "C-x b" (switch-buffer) is displaying the list of buffers (ordered by last use time). There is now way to list the buffers by order of creation (I guess this is how you want to cycle here).

I can add the support to cycle as you described, though I am a bit worried that it becomes hard to understand the buffer order. Maybe it was a wrong idea at first to sort them by use time in switch-buffer - or maybe I am overthinking and it won't be an issue to see M-n and M-p cycle differently than what C-x b shows;

Any idea/thought? Anyway I am looking to add the functions you asked. Just wondering about the whole picture here, and if you have ideas I am all hears.

**Comment by parkouss** [comment]
I have a working branch, see the associated PR. You can try it out and give me some feedback if you want to - I am planning to give it some more testing, then merge it if all looks good.

**Comment by bremner** [comment]
I've been testing this a bit today, and so far so good. I haven't been annoyed so far by the ordering of buffers, but to be honest I'm just starting to use C-x b. I did see one traceback, but I have no real reason to believe it has to do with the new buffer cycling. I'll attach the file just in case it's interesting.
[trace.txt](https://github.com/parkouss/webmacs/files/2321582/trace.txt)

**Comment by parkouss** [comment]
Ok, thanks for the traceback. I made some changes about the right_label code which is impacted, so let's wait to see if the issue come back when this PR is merged.

BTW finally I left the ordering in switch-buffer as it was. More exactly, I renamed it to switch-recent-buffer and bound it to the same bindings (C-x b, C-x C-b) so there should be no change from a user point of view. I added a new command switch-buffer which do the same but order buffers using the internal order.

I will merge the PR now, we'll see how it goes from there.
